version: "3.8"
services:
  app:
    container_name: wex
    image: purchase:0.1.0
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - 3060:3060
    volumes:
      - .:/app
    networks:
      - wex-network
    depends_on:
      db_wex:
        condition: service_healthy
    environment:
      - DATABASE_HOST=postgres://postgres:secret@db_wex:5432/purchase?sslmode=disable
      - HOST=0.0.0.0
  
  db_wex:
    container_name: db_wex
    image: postgres:14.2-alpine
    restart: always
    environment:
      - POSTGRES_DB=purchase
      - POSTGRES_PASSWORD=secret
      - POSTGRES_USER=postgres
    ports:
      - 5432:5432
    volumes:
      - vdb_wex:/var/lib/postgresql/data/
      - ./db/initdb-postgres.sql:/docker-entrypoint-initdb.d/initdb-postgres.sql
    networks:
      - wex-network
    healthcheck:
      test:
        [
          "CMD",
          "pg_isready",
          "-q",
          "-d",
          "purchase",
          "-U",
          "postgres"
        ]
      timeout: 20s
      interval: 2s
      retries: 10

networks:
  wex-network:
    name: wex-network
    driver: bridge

volumes:
    vdb_wex:
